src/sentry/api/endpoints/group_current_release.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+
+import sentry_sdk
+from sentry.api.base import EnvironmentMixin
+from sentry.api.bases import GroupEndpoint
+from sentry.api.helpers.environments import get_environments
+from sentry.api.serializers import serialize
+from sentry.api.serializers.models.grouprelease import GroupReleaseWithStatsSerializer
+from sentry.models import GroupRelease, ReleaseEnvironment, ReleaseProject
+
+
+class GroupCurrentReleaseEndpoint(GroupEndpoint, EnvironmentMixin):
+    def _get_current_release(self, group, environments):
+        release_projects = ReleaseProject.objects.filter(project_id=group.project_id).values_list(
```

Comment:
In case it's not obvious, this is the same code that was removed from `group_details.py`, with a slight refactor for readability but functionally the same query.

I tried several other ways of optimizing this but I don't think any of them used `Subquery`. Some of my experiments involved hand-tuned raw SQL queries, some of which would I think do something similar.

It sounds like a good idea, but I would rather kick it forward to a separate investigation task.

Reply:
Doing it separately sounds good to me. The old code was likely not using Subquery() as we were likely on old django that didn't support that behavior yet.